server/src/diagnostic_ext.rs

```diff
@@ -55,28 +65,38 @@ pub async fn compile_and_publish_diagnostics(
         // Trigger a compilation and get any diagnostics that were reported during it.
         let diagnostics = configuration_set.trigger_compilation();
         // Publish those diagnostics.
-        publish_diagnostics_for_set(client, diagnostics, configuration_set).await;
+        publish_diagnostics_for_set(backend, diagnostics, configuration_set).await;
     }
 }
 
 /// Processes a list of diagnostics and updates the publish map with LSP-compatible diagnostics.
 ///
 /// This function filters out any diagnostics that do not have a span or cannot be converted
 /// to an LSP diagnostic. It then updates the given publish map with the processed diagnostics.
+/// Any diagnostics that do not have a span are returned for further processing.
 pub fn process_diagnostics(
     diagnostics: Vec<slicec::diagnostics::Diagnostic>,
     publish_map: &mut HashMap<Url, Vec<tower_lsp::lsp_types::Diagnostic>>,
-) {
+) -> Vec<slicec::diagnostics::Diagnostic> {
+    let mut spanless_diagnostics = Vec::new();
     diagnostics
         .into_iter()
         .filter_map(|diagnostic| {
+            // The empty span case is handled by the `try_into_lsp_diagnostic` function.
```

Comment:
I'm mostly certain this comment is wrong.
`diagnostic.span()?` just immediately returns.
At least from the closure, maybe from the whole function too?